frontend/src/components/assessments/student-experience/question-components/MultiSelect.tsx

```diff
@@ -1,16 +1,46 @@
-import React from "react";
+import React, { useContext, useMemo } from "react";
 import { Checkbox, CheckboxGroup, VStack } from "@chakra-ui/react";
 
+import StudentContext from "../../../../contexts/StudentContext";
+import { answerValues, updatedAnswer } from "../../../../utils/StudentUtils";
+
 interface MultiSelectProps {
+  answerElementIndex: number;
   options: string[];
 }
-const MultiSelect = ({ options }: MultiSelectProps): React.ReactElement => {
+
+const MultiSelect = ({
+  answerElementIndex,
+  options,
+}: MultiSelectProps): React.ReactElement => {
+  const { currentQuestionIndex, answers, setAnswers } = useContext(
+    StudentContext,
+  );
+
+  const currentAnswer = useMemo(() => {
+    return answerValues(currentQuestionIndex, answerElementIndex, answers);
+  }, [currentQuestionIndex, answers, answerElementIndex]);
+
+  const updateAnswer = (value: number[]) => {
+    setAnswers((prevAnswers) => {
+      return updatedAnswer(
+        answerElementIndex,
+        currentQuestionIndex,
+        value,
+        prevAnswers,
+      );
+    });
+  };
```

Comment:
I don't love how all 3 of these question-components repeat this `currentAnswer` and `updateAnswer`. not sure the best way to refactor this logic out though, so let me know if you have suggestions!

Reply:
Seems like a custom hook kind of deal! Like `useAnswerState` taking in the `answerElementIndex` and an optional validator, maybe?

Reply:
something to look forward to for tomorrow 😍